executor/include/executor/best_fit.hpp

```diff
@@ -0,0 +1,111 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2014-, Open Perception, Inc.
+ *  Author(s): Shrijit Singh <shrijitsingh99@gmail.com>
+ *
+ */
+
+#pragma once
+
+#include <executor/type_trait.h>
+
+#include <executor/default/cuda_executor.hpp>
+#include <executor/default/inline_executor.hpp>
+#include <executor/default/omp_executor.hpp>
+#include <executor/default/sse_executor.hpp>
+
+namespace executor {
+
+static const auto best_fit_executors =
+    std::make_tuple(executor::omp_executor<>{}, executor::sse_executor<>{},
+                    executor::inline_executor<>{});
+
+struct executor_runtime_checks {
+  template <typename Executor, typename executor::instance_of_base<
+                                   Executor, executor::inline_executor> = 0>
+  static bool check(Executor& exec) {
+    return true;
+  }
+
+  template <typename Executor, typename executor::instance_of_base<
+                                   Executor, executor::sse_executor> = 0>
+  static bool check(Executor& exec) {
+    return true;
+  }
+
+  template <typename Executor, typename executor::instance_of_base<
+                                   Executor, executor::omp_executor> = 0>
+  static bool check(Executor& exec) {
+    return true;
+  }
+
+  template <typename Executor, typename executor::instance_of_base<
+                                   Executor, executor::cuda_executor> = 0>
+  static bool check(Executor& exec) {
+    return true;
+  }
+};
```

Comment:
So currently we don't have any runtime checks but since we have decided some runtime checks they will go here and return true/false based on the runtime conditions.
https://github.com/shrijitsingh99/unified-executors/issues/22